### ISSUE TITLE
java.lang.ClassCastException: java.base/java.lang.String cannot be cast to java.base/java.util.UUID when using UUID types with MockFileDatabase

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#8123]
When using a UUID column in PostgreSQL or H2, such as:

```sql
create table person (
   id uuid primary key,
   name text
);

insert into person (id, name) values (cast('072c6ba1-f06d-4c50-807c-e6b8e84a6e5b' as uuid), 'X');
```

And then mocking that column using the `MockFileDatabase` (which can only provide `String` content):

```java
    @Test
    public void testMockFileDatabaseDao() throws IOException {
        PersonDao dao = new PersonDao(new MockConfiguration(ctx.configuration(), new MockFileDatabase(
            "select .*;\n" +
            "> ID                                     NAME\n" +
            "> ------------------------------------- -----------------\n" +
            "> efb2d576-f2d3-4a8e-9715-dc95fdd14722   eddie\n" +
            "@ rows: 1"
        )));

        test(dao);
    }

    private void test(PersonDao dao) {
        Person person = dao.fetchOneById(uuid);
        assertEquals(uuid, person.getId());
        assertEquals("X", person.getName());
    }
```

Then we get the following exception:

```
org.jooq.exception.DataAccessException: SQL [select "PERSON"."ID", "PERSON"."NAME" from "PERSON" where "PERSON"."ID" = cast(? as uuid) -- SQL rendered with a free trial version of jOOQ 3.12.0-SNAPSHOT]; Error while reading field: "PERSON"."ID", at JDBC index: 1
	at org.jooq_3.12.0-SNAPSHOT.H2.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:2457)
	at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:818)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1623)
	at org.jooq.impl.CursorImpl$CursorIterator.hasNext(CursorImpl.java:1574)
	at org.jooq.impl.CursorImpl.fetchNext(CursorImpl.java:346)
	at org.jooq.impl.Tools.fetchOne(Tools.java:1824)
	at org.jooq.impl.AbstractResultQuery.fetchOne(AbstractResultQuery.java:551)
	at org.jooq.impl.SelectImpl.fetchOne(SelectImpl.java:2791)
	at org.jooq.impl.DAOImpl.fetchOne(DAOImpl.java:320)
	at org.jooq.codegen.test.mockuuidsindaos.db.tables.daos.PersonDao.fetchOneById(PersonDao.java:64)
	at org.jooq.codegen.test.mockuuidsindaos.MockUuidsInDaosTest.test(MockUuidsInDaosTest.java:63)
	at org.jooq.codegen.test.mockuuidsindaos.MockUuidsInDaosTest.testMockFileDatabaseDao(MockUuidsInDaosTest.java:99)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:541)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:763)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:463)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:209)
Caused by: java.sql.SQLException: Error while reading field: "PERSON"."ID", at JDBC index: 1
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1719)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.operate(CursorImpl.java:1678)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.operate(CursorImpl.java:1)
	at org.jooq.impl.RecordDelegate.operate(RecordDelegate.java:125)
	at org.jooq.impl.CursorImpl$CursorIterator.fetchNext(CursorImpl.java:1607)
	... 34 more
Caused by: java.lang.ClassCastException: java.base/java.lang.String cannot be cast to java.base/java.util.UUID
	at org.jooq.impl.DefaultBinding$DefaultUUIDBinding.get0(DefaultBinding.java:4005)
	at org.jooq.impl.DefaultBinding$DefaultUUIDBinding.get0(DefaultBinding.java:1)
	at org.jooq.impl.DefaultBinding$AbstractBinding.get(DefaultBinding.java:797)
	at org.jooq.impl.CursorImpl$CursorIterator$CursorRecordInitialiser.setValue(CursorImpl.java:1709)
	... 38 more
```

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport scheduled for 3.11.8 (#8127)

**Comment by swathistudyblue** [comment]
Thank you for addressing this. This means, the issue should be fixed in jooq 3.12, right? But I still see the same issue? Any insights?

**Comment by lukaseder** [comment]
Yes, this is fixed in jOOQ 3.12. What do you mean by "I still see the same issue"? Did you build jOOQ 3.12 and try it again?
